File: sust-ims-ui/src/main/resources/charts/pie-chart.js
/**
 *
 * {
			  labelInterpolationFnc: function(value) {
				    return Math.round(value / data.series.reduce(sum) * 100) + '%';  
			  }
 */
var charts_PieChart		= function(){
	this.onStateChange	= function(){
		//tools
		var sum 			= function(a, b) { return a + b };
		var showPrecentage	= this.getState().showPrecentage;
		
		var	data		= {series: this.getState().series};
		var options 	= {
				labelInterpolationFnc:
				function(value) {return Math.round(value / data.series.reduce(sum) * 100) + '%';}
			   ,showLabel: this.getState().showLabel
			   ,donut:	   this.getState().donut
			   ,donutWidth:this.getState().donutWidth
			   ,startAngle:this.getState().startAngle
			   ,total:	   this.getState().total
		};
		
		if(showPrecentage === true)
		{var	PieChart	= new Chartist.Pie('.pie-chart', data, options);}
		else
		{
			var	data		= {labels: this.getState().labels, series: this.getState().series};
			var options 	= {
					labelInterpolationFnc:
					function(value) {return value}
				   ,showLabel: this.getState().showLabel
				   ,donut	 : this.getState().donut
				   ,donutWidth:this.getState().donutWidth
				   ,startAngle:this.getState().startAngle
				   ,total:	   this.getState().total
				   
			};
			var	PieChart	= new Chartist.Pie('.pie-chart', data, options);
		}
		
	}
}

File: sust-ims-ui/src/main/java/charts/PieChart.java
package charts;

import java.util.List;

public class PieChart<T, U> extends PieChartComponent{

	/**
	 * 
	 */
	private static final long serialVersionUID = -8804166587358978074L;
	public PieChart(){
		
	}
	
	private static final String CHART_CLASS		 =  "ct-chart"
			   				   ,PIE_CHART_CLASS	 =  "pie-chart";
	{
		init();	
	}
	private void init(){
		setStyleName(PIE_CHART_CLASS);
		addStyleName(CHART_CLASS);	
	}
	@SuppressWarnings("unchecked")
	@Override
	public PieChartState<T, U> getState(){
		return (PieChartState<T, U>) super.getState();
	}
	public void setLabels(List<T> labels){
		getState().labels	= labels; 
	}
	public List<T> getLabels(){
		return getState().labels;  
	}
	public void setSerises(List<U> serises){
		getState().series	= serises;
	}
	public  List<U> getSerises(){
		return  getState().series; 
	}
	public void setShowPrecentageEnabled(boolean showPrecentageEnabled){
		getState().showPrecentage	= showPrecentageEnabled;
	}
	public boolean isShowPrecentageEnabled(){
		return getState().showPrecentage;
	}
	public void setChartLabelsVisible(boolean showLabel){
		getState().showLabel	= showLabel;
	}
	public boolean isChartLabelsVisible(){
		return	getState().showLabel;
	}
	public void setDonutEnabled(boolean isEnabled){
		getState().donut	= isEnabled;
	}
	public boolean	isDonutEnabled(){
		return getState().donut;
	}
	public void setDonutWidth(int donutWidth){
		getState().donutWidth	= donutWidth;
	}
	public int	getDonutWidth(){
		return	getState().donutWidth;
	}
	public void setStartAngle(int startAngle){
		getState().startAngle	= startAngle;
	}
	public int  getStartAngle(){
		return  getState().startAngle;
	}
	public void setTotal(int total){
		getState().total	= total;
	}
	public int	getTotal(){
		return	getState().total;
	}
	
}